ragna/assistants/_api.py

```diff
@@ -16,22 +17,28 @@ def requirements(cls) -> list[Requirement]:
         return [EnvVarRequirement(cls._API_KEY_ENV_VAR)]
 
     def __init__(self) -> None:
-        self._client = httpx.AsyncClient(
+        self._api_key = os.environ[self._API_KEY_ENV_VAR]
+
+        kwargs: dict[str, Any] = dict(
             headers={"User-Agent": f"{ragna.__version__}/{self}"},
             timeout=60,
         )
-        self._api_key = os.environ[self._API_KEY_ENV_VAR]
+        self._sync_client = httpx.Client(**kwargs)
+        self._async_client = httpx.AsyncClient(**kwargs)
 
     async def answer(
         self, prompt: str, sources: list[Source], *, max_new_tokens: int = 256
     ) -> AsyncIterator[str]:
-        async for chunk in self._call_api(  # type: ignore[attr-defined, misc]
-            prompt, sources, max_new_tokens=max_new_tokens
+        async for chunk in as_async_iterator(
+            self._call_api,
+            prompt,
+            sources,
+            max_new_tokens=max_new_tokens,
         ):
             yield chunk
 
     @abc.abstractmethod
-    async def _call_api(
+    def _call_api(
```

Comment:
This makes no functional difference. We just use the sync prototype in other places when the function can be overwritten as sync or async.